lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,248 @@
+use nalgebra::zero;
+use crate::traits::HasPosition;
+
+// ======================
+// === DragProperties ===
+// ======================
+
+/// This structure contains air dragging properties.
+#[derive(Default, Clone)]
+pub struct DragProperties {
+    amount : Rc<RefCell<f32>>
+}
+
+impl DragProperties {
+    pub fn new(amount:f32) -> Self {
+        let amount = Rc::new(RefCell::new(amount));
+        Self { amount }
+    }
+
+    /// Gets dragging amount.
+    pub fn amount(&self) -> f32 { *self.amount.borrow() }
+
+    /// Sets dragging amount.
+    pub fn set_amount(&mut self, amount:f32) {
+        *self.amount.borrow_mut() = amount
+    }
+}
+
+// ========================
+// === SpringProperties ===
+// ========================
+
+struct SpringData {
+    mass        : f32,
+    coefficient : f32,
+    fixed_point : Vector3<f32>
+}
+
+/// This structure contains spring physics properties.
+#[derive(Clone)]
+pub struct SpringProperties {
+    data : Rc<RefCell<SpringData>>
+}
+
+impl Default for SpringProperties {
+    fn default() -> Self {
+        Self::new(zero(), zero(), zero())
+    }
+}
+
+impl SpringProperties {
+    pub fn new(mass:f32, coefficient:f32, fixed_point:Vector3<f32>) -> Self {
+        let data = Rc::new(RefCell::new(SpringData { mass, coefficient, fixed_point }));
+        Self { data }
+    }
+
+    /// Gets mass.
+    pub fn mass(&self) -> f32 {
+        self.data.borrow().mass
+    }
+
+    /// Sets mass.
+    pub fn set_mass(&mut self, mass:f32) {
+        self.data.borrow_mut().mass = mass
+    }
+
+    /// Gets coefficient.
+    pub fn coefficient(&self) -> f32 {
+        self.data.borrow().coefficient
+    }
+
+    /// Sets coefficient.
+    pub fn set_coefficient(&mut self, coefficient:f32) {
+        self.data.borrow_mut().coefficient = coefficient
+    }
+
+    /// Gets spring's fixed point.
+    pub fn fixed_point(&self) -> Vector3<f32> {
+        self.data.borrow().fixed_point
+    }
+
+    /// Sets spring's fixed point.
+    pub fn set_fixed_point(&mut self, fixed_point:Vector3<f32>) {
+        self.data.borrow_mut().fixed_point = fixed_point
+    }
+}
+
+// ============================
+// === KinematicProperties ===
+// ============================
+
+struct KinematicData {
+    position     : Vector3<f32>,
+    velocity     : Vector3<f32>,
+    acceleration : Vector3<f32>
+}
+
+/// This structure contains kinematics properties.
+#[derive(Clone)]
+pub struct KinematicProperties {
+    data : Rc<RefCell<KinematicData>>
+}
+
+impl Default for KinematicProperties {
+    fn default() -> Self { Self::new(zero(), zero(), zero()) }
+}
+
+impl KinematicProperties {
+    pub fn new(position:Vector3<f32>, velocity:Vector3<f32>, acceleration:Vector3<f32>) -> Self {
+        let data = Rc::new(RefCell::new(KinematicData { position, velocity, acceleration }));
+        Self { data }
+    }
+
+    /// Gets velocity.
+    pub fn velocity(&self) -> Vector3<f32> {
+        self.data.borrow().velocity
+    }
+
```

Comment:
We agreed some time ago that we dont need to put docs on getters and setters cause it is just stupid and doesnt bring any more information and jsut takes your time. Just fyi. However, please then refactor getters and setters so:

```rust
// === Getters ===

impl KinematicProperties {
    ...
}


// === Setters ===

impl KinematicProperties {
    ...
}
```